## `polyploid-genotyping`

[![Build Status](https://travis-ci.org/pblischak/polyploid-genotyping.svg?branch=master)](https://travis-ci.org/pblischak/polyploid-genotyping)[![DOI](https://zenodo.org/badge/75424744.svg)](https://zenodo.org/badge/latestdoi/75424744)

This is the GitHub repository for the manuscript "SNP genotyping and parameter estimation in polyploids using low-coverage sequencing data," which is currently under peer review. We have also posted a preprint on <a href="http://biorxiv.org/content/early/2017/03/26/120261" target="_blank">bioRxiv</a>.

 - `ebg/`: C++ source code for EM/ECM algorithms for genotyping with our models.

 - `helper-scripts/`: Perl, R, and bash script for extracting and filtering allele depth info from a VCF file for use with `ebg`.

 - `Rcode/`: R and C++ code for performing simulations and for running analyses.

 - `data/`: example data sets from *Betula pendula*, *B*. *pubescens*, and *Andropogon gerardii* that were used in the manuscript.

 - `docs/`: Rmd and HTML files for GitHub pages site.


More details on the contents of each folder can be found in their respective README files.
